adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepositoryTest.java

```diff
@@ -447,4 +445,108 @@ void addFareRuleAndGetFareRuleShouldReturnSameEntity() {
         assertEquals(mockFareRule01, underTest.getFareRule("fare id1", null, null,
                 null, null));
     }
-}
\ No newline at end of file
+
+    @Test
+    void addTwiceSameShapePointShouldReturnNull() {
+        final ShapePoint mockShapePoint = mock(ShapePoint.class);
+        when(mockShapePoint.getShapeId()).thenReturn("test id");
+        when(mockShapePoint.getShapePtLat()).thenReturn(50f);
+        when(mockShapePoint.getShapePtLon()).thenReturn(100f);
+        when(mockShapePoint.getShapePtSequence()).thenReturn(4);
+        when(mockShapePoint.getShapeDistTraveled()).thenReturn(56f);
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+
+        underTest.addShapePoint(mockShapePoint);
```

Comment:
I would also pull the shape you just inserted into the repo back out again and then assert that the field values for the shape equals the one you put in - that way it's an end-to-end test.